causing/utils.py

```diff
@@ -348,7 +348,9 @@ def vecmat(mz):
 
 def render_dot(dot_str, filename):
     """render Graphviz graph from dot_str to svg using pydot"""
-    pydot.graph_from_dot_data(dot_str)[0].write_svg(filename)
+    data = pydot.graph_from_dot_data(dot_str)
```

Comment:
> And if we have empty nodes, can't we stop generating them instead of removing them afterwards?

The `dot_str` before this point does not consist of any newline. It is only added after executing `graph_from_dot_data` function.